### PR TITLE
[lipstick] Don't deference invalid notifications. Fixes JB#30814

### PR DESCRIPTION
**Pull request by rainemak**

### CODE REVIEW

**Comment by matthewvogt** [comment]
Looks good.

I think the iterator dereferences in the following functions could also be checked, even if not required to fix the issue:
 - NotificationManager::removeNotificationsWithCategory(const QString &category)
 - NotificationManager::updateNotificationsWithCategory(const QString &category)
 - NotificationManager::removeUserRemovableNotifications()

**Comment by rainemak** [comment]
Will add.

**Comment by rainemak** [comment]
Fixed review comments.

File: src/notifications/notificationmanager.cpp [hunk 1]
```diff
@@ -414,7 +414,7 @@ NotificationList NotificationManager::GetNotifications(const QString &owner)
     QHash<uint, LipstickNotification *>::const_iterator it = notifications.constBegin(), end = notifications.constEnd();
     for ( ; it != end; ++it) {
         LipstickNotification *notification = it.value();
-        if (notification->owner() == owner) {
+        if (notification && notification->owner() == owner) {
```
Comment (pvuorela) [inline src/notifications/notificationmanager.cpp]: Wouldn't this be working around the problem, like would expect QHash not to ever contain null pointer values?

Comment (pvuorela) [inline src/notifications/notificationmanager.cpp]: ...if there's a problem to begin with in this kind of places. Seems only newly created instances are inserted. Could be that only bug was trying to delete with non-existent id?

Comment (rainemak) [inline src/notifications/notificationmanager.cpp]: Kind of working around. Debugged quite extensively and didn't find the root where the invalid id gets inserted. Enable debugs and see take lines at 343 and 370.

Feel free to propose the proper fix or fix by yourself.

**Pull request closed without merging**